unit_scaling/functional.py

```diff
@@ -252,6 +276,47 @@ def embedding(
     )
 
 
+if hasattr(F, "scaled_dot_product_attention"):
+
+    @docstring_from(
+        F.scaled_dot_product_attention,
+        short_description=(
+            "A **unit-scaled** dot-product attention function. Note that this will use"
+            "whatever underlying PyTorch scaled_dot_product_attention implementation"
+            "is available, so if flash attention is enabled it will be used here too."
+            "\n\n"
+            "Computes scaled dot product attention on query, key and value tensors,"
+            "using an optional attention mask if passed, and applying dropout if a"
+            "probability greater than 0.0 is specified."
+        ),
+    )
+    def scaled_dot_product_attention(
+        query: Tensor,
+        key: Tensor,
+        value: Tensor,
+        attn_mask: Optional[Tensor] = None,
+        dropout_p: float = 0.0,
+        is_causal: bool = False,
+        scale: Optional[float] = None,
```

Comment:
What is the unused `scale` arg for? I can't see it in `torch.scaled_dot_product_attention`.